c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2718,6 +2719,9 @@ class DeviceCachingAllocator {
     } else if (
         CUDAAllocatorConfig::expandable_segments() &&
         !(in_fbcode && p.pool->owner_PrivatePool)) {
+      TORCH_CHECK(
+          !active_pool,
```

Comment:
Is this check overly restrictive? I think it still works with custom memory pools as long as they don't have custom allocators. I.e. should it be `!active_pool || !active_pool->allocator()`?

there are cases when one may use explicit memory pools without custom allocators to reduce memory fragmentation

Reply:
It is exactly what you say it should be https://github.com/pytorch/pytorch/blob/1fc010a9d8ea95bb74e54b31d17eba56ef16c27c/c10/cuda/CUDACachingAllocator.cpp#L2865-L2876